packages/expo-video/src/VideoView.tsx

```diff
@@ -1,18 +1,24 @@
-import { ReactNode, PureComponent, useMemo, createRef } from 'react';
+import {
+  ReactNode,
+  PureComponent,
+  DependencyList,
+  createRef,
+  useRef,
+  useMemo,
+  useEffect,
+} from 'react';
 
 import NativeVideoModule from './NativeVideoModule';
 import NativeVideoView from './NativeVideoView';
 import { VideoPlayer, VideoSource, VideoViewProps } from './VideoView.types';
 
 export function useVideoPlayer(source: VideoSource): VideoPlayer {
-  return useMemo(() => {
-    if (typeof source === 'string') {
-      return new NativeVideoModule.VideoPlayer({
-        uri: source,
-      });
-    }
-    return new NativeVideoModule.VideoPlayer(source);
-  }, []);
+  const parsedSource = typeof source === 'string' ? { uri: source } : source;
+
+  return useReleasingSharedObject(
+    () => new NativeVideoModule.VideoPlayer(parsedSource),
+    [parsedSource?.uri, ...Object.values(parsedSource?.drm ?? {})]
```

Comment:
This also doesn't seem to be correct as the `drm` has optional properties so the number of values may change which is, as far as I know, not allowed in hook dependencies. Also, `headers` is an object as well 😅 
I'm afraid we'll need to use `JSON.stringify(parsedSource)`, luckily the source object is not big and nested so that operation shouldn't be too expensify.